src/_pytest/mark/legacy.py

```diff
@@ -84,8 +84,13 @@ def matchmark(colitem, markexpr):
     """Tries to match on any marker names, attached to the given colitem."""
     try:
         return eval(markexpr, {}, MarkMapping.from_item(colitem))
-    except SyntaxError as e:
-        raise SyntaxError(str(e) + "\nMarker expression must be valid Python!")
+    except (SyntaxError, TypeError):
```

Comment:
I'd prefer to match `matchkeywords` (function one below this one), namely catch *all* errors (`Exception`) instead of specific ones.

We should also be consistent in the error message. I prefer the one in `matchkeyword` just because it is shorter and clear enough IMO.

Reply:
sure thing!

Reply:
thanks @bluetech noted the discussion linked, and also agree I think its ok to improve around this (if its not the fully fledged solution just (yet))